pkg/system/phase4_configuring.go

```diff
@@ -677,6 +687,104 @@ func (r *Reconciler) prepareGCPBackingStore() error {
 	return nil
 }
 
+func (r *Reconciler) prepareIBMBackingStore() error {
+	r.Logger.Info("Preparing backing store in IBM Cloud")
+
+	var (
+		endpoint string
+		location string
+	)
+
+	cloudCredsSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.IBMCloudCreds.Spec.SecretRef.Name,
+			Namespace: r.IBMCloudCreds.Spec.SecretRef.Namespace,
+		},
+	}
+
+	util.KubeCheck(cloudCredsSecret)
+	if cloudCredsSecret.UID == "" {
+		// Secret not found
+		r.Logger.Errorf("Cloud credentials secret %q is not ready yet", r.IBMCloudCreds.Spec.SecretRef.Name)
+		return fmt.Errorf("Cloud credentials secret %q is not ready yet", r.IBMCloudCreds.Spec.SecretRef.Name)
+	}
+
+	if val, ok := cloudCredsSecret.StringData["IBM_COS_Endpoint"]; ok {
+		// Use the endpoint provided in the secret
+		endpoint = val
+		r.Logger.Infof("Endpoint provided in secret: %q", endpoint)
+		if val, ok := cloudCredsSecret.StringData["IBM_COS_Location"]; ok {
+			location = val
```

Comment:
Any reason not to add logging for `location` as well?
I would maybe even log them later on after the assignment.

Reply:
Update the code to log `location` as well.